File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1]
        for i in range(1, len(nums)):
            res.append(nums[i - 1] * res[-1])
        
        temp = nums[-1]
        for i in range(len(nums) - 2, -1, -1):
            res[i] = res[i] * temp
            temp *= nums[i]
        
        return res

File: Data Structures & Algorithms/non-overlapping-intervals/submission-0.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        int n = intervals.size();
        
        int res = 0;
        int temp = intervals[0][1];
        for(int i = 1; i < n; i++) {
            if (intervals[i][0] < temp) {
                res += 1;
                temp = min(temp, intervals[i][1]);
            }
            else
                temp = intervals[i][1];
        }
        return res;
    }
};
